### DISCUSSION TITLE
Support for MinGW cross-compilation?

### DISCUSSION THREAD

**Discussion by Semphriss** [SuperTux/supertux#1843] [Ideas]
Should we support MinGW compilation from Linux for Windows?

Having all builds possible from Linux would simplify custom builds on servers.

**Comment by tobbi** [comment]
There's quite a cross-compilation in this video: https://www.youtube.com/watch?v=SJUhlRoBL8M

**Comment by Zwatotem** [comment]
Support is always fine, however I wouldn't rely on having Windows builds done on Linux as it may make possible companion workflows (e.g. automatic tests, performance metrics) harder to work with.
I assume that there's zero difference in final exec, compilation time may be lower. If it's just a question of changing a flag in CMake it may be worth it. If it takes more, I don't think so.
Apart from this case, I don't think there's someone specifically interested in being able to compile and run on different environments.

Grzegorz Przybylski
________________________________
From: Semphriss ***@***.***>
Sent: Friday, October 8, 2021 11:00:29 PM
To: SuperTux/supertux ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [SuperTux/supertux] Support for MinGW cross-compilation? (Discussion #1843)


Should we support MinGW compilation from Linux for Windows?

Having all builds possible from Linux would simplify custom builds on servers.

**Reply by Semphriss** [reply]
I've compiled multiple programs with MinGW and they've been working well; SuperTux has a slightly messier buildsystem (rip CMakeLists.txt) which makes it slightly more difficult to support cross-compilation.

Vcpkg has community support for MinGW; all dependencies [except OpenAL](https://github.com/microsoft/vcpkg/issues/20539) build properly. If we omit that, I think it might be possible to cross-compile SuperTux with as much stability as native builds.

Cross-compilation is interesting for me as it allows me to build for more targets from Linux. I wanted to make my own buildsystem using my devices, and I don't have a stable enough budget to pay for Windows servers; MinGW builds could save me a lot of time.
